fix some plpgsql errors

MGorkov opened the pull request:
This patch fixes the [#243](https://github.com/pganalyze/libpg_query/issues/243) , [#160](https://github.com/pganalyze/libpg_query/issues/160) , [#248](https://github.com/pganalyze/libpg_query/issues/248) issues and the following errors:
"variable "$1" does not exist"
"collations are not supported by type -"
"variable "ref" must be of type cursor or refcursor"
missing expression at or near ";"

changes in JSON:
1. add declared variable with "alliasfor" parameter pointing to associated variable, e.g. DECLARE _a ALIAS FOR $1;
```
    {
      PLpgSQL_var: {
        refname: '_a',
        aliasfor: '$1',
        lineno: 3,
        datatype: { PLpgSQL_type: { typname: 'text' } }
      }
```
2. add "retvarno" parameter for RETURN NEXT statement
```
    {
      PLpgSQL_stmt_return_next: { lineno: 6, retvarno: 2 }
    }
```

MGorkov committed:
fix ival for negatives and zeros

lfittl commented:
Thanks for the patch, and conceptually looks good!

However it appears there is a problem with the `protobuf-cpp` tests here, blocking a merge.

I believe this is actually a larger issue with how we handle `oneof` in our JSON output, I started fixing that in #246, pulling in your commit there as well. Assuming I can work out the open issues on that PR I'll close this here and fix it by merging the other PR.

MGorkov committed:
fix unnamed params

MGorkov committed:
fix setting typoid for refcursor and cursor data types

MGorkov committed:
fix plpgsql parser errors:
- variable "$1" does not exist
- collations are not supported by type -
- variable "ref" must be of type cursor or refcursor

MGorkov committed:
fix plpgsql error 'missing expression at or near ";"' when parsing RETURN NEXT with no expression

MGorkov committed:
add retvarno for return next

MGorkov committed:
add aliasfor

MGorkov committed:
add %rowtype support

lfittl commented:
@MGorkov Thanks for your contributions in this PR! I ended up reworking them a bit and merging them in #263.

I did not take all changes, but tried to keep it more narrow to the parts that looked most relevant to make parsing work in cases where it didn't work before.

If there are additional changes you'd like to see, could you propose these in a new PR? I'll close this here for now, since I believe the main issues are addressed.
